manifests/service.pp

```diff
@@ -7,11 +7,35 @@
 #
 # * Justin Lambert <mailto:jlambert@letsevenup.com>
 #
-class kibana::service {
+class kibana::service (
+  $service_provider    = $::kibana::params::service_provider,
+) {
+
+  if $service_provider == 'init' {
```

Comment:
Removing the duplication is way better.  I still think it goes in the install.pp though, mostly when I think about the concept of what should be notified when notifying a service - I don't think it's applicable to have the init script notified then, it should be done "once" as part of an install (just like installing a package installs an init script) and the service class just manages the service.